Automattic/wp-calypso
# Assistance migration modal: Navigate to checkout

## Proposed Changes

* This PR sends the user to the checkout page when accepting the migration deal, by accepting it, the user should see the "Assisted free site migration" product in the checkout, this was handled in https://github.com/Automattic/wp-calypso/pull/89906


<img width="270" alt="image" src="https://github.com/Automattic/wp-calypso/assets/2653810/36093cbb-780d-4b34-992a-bd5257b8a5a7">

## Testing Instructions



According to this pcmemI-2sX-p2 great post, I have mapped some flows using this step and furthermore should show the modal when navigating back, they are:

* site-setup
* site-migration-flow
* import-hosted-sites
* import-flow


Testing steps
* Pull and run this branch or use live link
* Navigate to the flows, adapting the URL with live link or calypso.localhost.
* Make sure to add the feature flag in the URL to be able to see the modal when navigating back `flags=migration_assistance_modal`
* When accepting the deal, the user should be taken to the checkout 
* Navigating back should point the user to the previous step
* The checkout should have "Assisted free site migration" product


## Pre-merge Checklist
